File: src/main/java/p00/Problem_0029.java
package p00;

import java.math.BigInteger;
import java.util.HashSet;

public class Problem_0029 {

    public static int solve() {

        HashSet<BigInteger> sorted = new HashSet<>();

        for (int a = 2; a <= 100; a++) {
            BigInteger number = BigInteger.valueOf(a);
            for (int b = 2; b <= 100; b++) {
                number = number.multiply(BigInteger.valueOf(a));
                sorted.add(number);
            }
        }

        return sorted.size();
    }

}

File: src/main/kotlin/p00/Problem0029.kt
package p00

import java.math.BigInteger
import java.util.*

class Problem0029 {
    fun solve(): Int {
        val sorted = HashSet<BigInteger>()
        for (a in 2..100) {
            var number = BigInteger.valueOf(a.toLong())
            for (b in 2..100) {
                number = number.multiply(BigInteger.valueOf(a.toLong()))
                sorted.add(number)
            }
        }
        return sorted.size
    }
}